src/vs/workbench/contrib/terminal/browser/links/terminalWordLinkDetector.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { ITerminalSimpleLink, ITerminalLinkDetector, TerminalLinkType } from 'vs/workbench/contrib/terminal/browser/links/links';
+import { convertLinkRangeToBuffer, getXtermLineContent } from 'vs/workbench/contrib/terminal/browser/links/terminalLinkHelpers';
+import { ITerminalConfiguration, TERMINAL_CONFIG_SECTION } from 'vs/workbench/contrib/terminal/common/terminal';
+import { IBufferLine, Terminal } from 'xterm';
+
+const enum Constants {
+	/**
+	 * The max line length to try extract word links from.
+	 */
+	MaxLineLength = 2000
```

Comment:
there's only one, so should this enum not exist?

Reply:
Inlining the number would mean we have a magic number that needs a comment to explain it, a const enum means we can document it up the top and have a named constant that is completely compiled out of the js output